### DISCUSSION TITLE
Uncaught Error: Disallowed operation called within global scope

### DISCUSSION THREAD

**Discussion by brachkow** [rhinobase/hono-rate-limiter#35] [Q&A] [answered]
I created a middleware:

emailRateLimiter:
```ts
import { rateLimiter } from 'hono-rate-limiter';

export const emailRateLimiter = rateLimiter({
  windowMs: 15 * 60 * 1000,
  limit: 5,
  standardHeaders: 'draft-6',
  keyGenerator: (c) => c.req.url
});
```

then im trying to use it (in conjunction with `@hono/zod-openapi`).

```ts
app
  .openapi(
    createRoute({
      method: 'post',
      path: '/users/code',
      request: {
        // --
      },
      responses: {
       // --
      },
      middleware: [emailRateLimiter],
    }),
    async (c) => {
     // --
      }
    },
  )
```

which results in

```
Uncaught Error: Disallowed operation called within global scope. Asynchronous I/O (ex: fetch() or connect()), setting a timeout, and generating random values are not allowed within global scope. To fix this error, perform this operation within a handler. https://developers.cloudflare.com/workers/runtime-apis/handlers/
```

**Comment by MathurAditya724** [comment]
Interesting, I haven't tried it with `@hono/zod-openapi`, but okay give me some time to look into this.

**Comment by MathurAditya724** [comment, marked as answer]
The issue is you are running this in Cloudflare env, either `workers` or `pages`. You will have to pass a store as by default it uses a memory store that cannot run in Cloudflare. If you want to use Cloudflare infra as a store, you can look into this https://www.npmjs.com/package/@hono-rate-limiter/cloudflare.

**Reply by brachkow** [reply]
Thank you! Maybe you can add note about MemoryStore not working on CF workers? Hono is CF backed framework that used mostly with workers – it's not obvious now that default option will not work in that env.

**Reply by MathurAditya724** [reply]
Yes, I will do that
